lib/iris/fileformats/netcdf.py

```diff
@@ -1958,7 +1949,11 @@ def set_packing_ncattrs(cfvar):
             # Explicitly assign the fill_value, which will be the type default
             # in the case of an unmasked array.
             if packing is None:
-                fill_value = cube.fill_value
+                if hasattr(cube.lazy_data(), 'fill_value'):
+                    fill_value = cube.lazy_data().fill_value
+                else:
+                    fill_value = None
+
```

Comment:
I am concerned that in the example where we have lazy masked data, we aren't setting the fill_value as it would just be None. I am not sure what the fall out of this would be